src/main/java/GmailReader.java

```diff
@@ -0,0 +1,174 @@
+import com.sun.org.omg.CORBA.ExceptionDescription;
+
+import javax.mail.Flags;
+import javax.mail.Folder;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+public class GmailReader {
+    public ReaderConfig readerConfig;
+    public ReaderAuthentication readerAuthentication;
+
+
+
+    public GmailReader(ReaderConfig readerConfig, ReaderAuthentication readerAuthentication) {
+        this.readerConfig = readerConfig;
+        this.readerAuthentication = readerAuthentication;
+
+    }
+
+    public class longVacationException extends Exception {
+        public longVacationException(String s) {
+            System.out.println(s);
+        }
+    }
+
+    public class pastDateException extends Exception {
+        public pastDateException(String s) {
+            System.out.println(s);
+        }
+    }
+
+    public class invalidDateFormatException extends Exception {
+        public invalidDateFormatException(String s) {
+            System.out.println(s);
+        }
+    }
+    public class invalidDateInformationException extends Exception {
+        public invalidDateInformationException(String s) {
+            System.out.println(s);
+        }
+    }
+
+
+    public boolean subjectAndSeenChecker(Message msg) throws MessagingException {
+        return (msg.getSubject() != null) && (!msg.isSet(Flags.Flag.SEEN));
+    }
+
+    public boolean numberOfWordsRepetition(String str, String words, int count) {
+        return str.equals(words) && (count == 0);
+    }
+
+    public ArrayList<Date> dataParser(String start, String end) throws ParseException {
+        ArrayList<Date> datesOfVacation = new ArrayList<Date>();
+        String pattern = "dd/MM/yyyy";
+        SimpleDateFormat format = new SimpleDateFormat(pattern);
+        Date startVacation = format.parse(start);
+        Date endVacation = format.parse(end);
+        datesOfVacation.add(startVacation);
+        datesOfVacation.add(endVacation);
+        return datesOfVacation;
+    }
+
+    public String stringCleaner(String str) {
+        str = str.toLowerCase();
+        str = str.replace(",", "");
+        str = str.replace(".", "");
+        str = str.replace(";", "");
+        str = str.replace(":", "");
+
+        return str;
+    }
+
+    public String[] getDate(String description) throws invalidDateInformationException {
+        int count = 0;
+        int i = 0;
+        String[] allMatches = new String[2];
+        Matcher match =
+                Pattern.compile("(0[1-9]|[12][0-9]|3[01])[- /.](0[1-9]|1[012])[- /.](19|20)\\d\\d").matcher(description);
```

Comment:
write a static variable above to make the line shorter and more readable

e.g. EMAIL_MATCHER = "(0[1-9]|[12][0-9]|3[01])[- /.](0[1-9]|1[012])[- /.](19|20)\\d\\d"
>> Pattern.compile(EMAIL_MATCHER)..